packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart

```diff
@@ -12,228 +11,69 @@ const _keyNotFoundRecoveryMessage =
-      recoverySuggestion: AmplifyExceptionMessages.missingExceptionMessage,
+/// The exception thrown when cancel a controllable operation.
+const s3ControllableOperationCanceledException = StorageOperationCanceled(
+  'The operation has been canceled.',
+  recoverySuggestion:
+      'This is an expected exception when you call cancel() API on the storage operation object. You need to handle this exception to take further action on operation cancellation.',
+);
+
+/// Creates a [StorageKeyNotFoundException] with the [underlyingException].
+StorageKeyNotFoundException createKeyNotFoundException(
+  Exception underlyingException,
+) =>
+    StorageKeyNotFoundException(
+      'Key is not found.',
+      recoverySuggestion: _keyNotFoundRecoveryMessage,
+      underlyingException: underlyingException,
     );
-  }
-
-  /// Creates a [S3Exception] that represents an unexpected error
-  /// when service API call doesn't return a valid content-length header.
-  factory S3Exception.unexpectedContentLengthFromService() {
-    return const S3Exception(
-      'Service has not returned a valid content length for requested object.',
-      recoverySuggestion: AmplifyExceptionMessages.missingExceptionMessage,
-    );
-  }
-
-  /// Creates a [S3Exception] that represents an expected exception
-  /// thrown when invoke cancel() on an eligible storage operation object.
-  factory S3Exception.controllableOperationCanceled() {
-    return const S3Exception(
-      'The requested operation has been canceled.',
-      recoverySuggestion:
-          'This is an expected exception when you call cancel() API on the storage operation object. You need to handle this exception to take further action on operation cancellation.',
-    );
-  }
 
-  /// An exception thrown when the `body` returned from GetObject is unexpectedly null.
-  factory S3Exception.unexpectedGetObjectBody() {
-    return const S3Exception(
-      'Unexpected null body from GetObject.',
-      recoverySuggestion: AmplifyExceptionMessages.missingExceptionMessage,
-    );
-  }
-
-  /// An exception created when attempt to resume a canceled operation.
-  factory S3Exception.resumeCanceledOperation() {
-    return const S3Exception(
-      'The operation has been canceled and can\'t be resumed.',
-      recoverySuggestion:
-          'You can only resume an operation that is paused by calling `pause()`.',
-    );
-  }
-
-  /// An exception thrown when the service doesn't return a valid uploadId on a
-  /// successful `CreateMultipartUploadRequest`.
-  factory S3Exception.unexpectedMultipartUploadId() {
-    return const S3Exception(
-      'Create multipart upload request succeeded, but it did not return a valid uploadId.',
-      recoverySuggestion: AmplifyExceptionMessages.missingExceptionMessage,
-    );
-  }
+/// Creates a [NetworkException] with the [AWSHttpException].
+NetworkException createNetworkExceptionFromAWSHttpException(
+  AWSHttpException exception,
+) {
+  return NetworkException(
```

Comment:
Since NetworkException will be exact same between Analytics, Storage and perhaps other categories, can we specify the exact NetworkException text in core?

Reply:
[Added an extension to `AWSHttpException`](https://github.com/aws-amplify/amplify-flutter/pull/2781/commits/8925c5f93455136eaadb0c60077e2e46b9933e23) to create a `NetworkException` with consistent messaging. Please let me know if other alternatives are preferred.